### PR TITLE
FIX: Carret position is invalid if editing text not at the end

### LINKED ISSUE #64: BUG: Carret position is invalid if not editing at the end of the text

### PR DESCRIPTION
**Pull request by radimhalfar**
- also added section for defining UI Elements and demonstrate usage of styles on it
closes #64

### CODE REVIEW

File: Styles/Classes/UIElement+Extensions/UITextField+Styles.m [hunk 1]
```diff
@@ -132,10 +132,16 @@ - (void)updateStylesForState:(TextInputState)state {
 
 - (void)applyTextStyle:(TextStyle *)style {
     NSString *text = self.text;
-    if (!text || !style) {
+
+	if (!text || !style) {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextField+Styles.m]: align

File: Styles/Classes/UIElement+Extensions/UITextField+Styles.m [hunk 2]
```diff
@@ -132,10 +132,16 @@ - (void)updateStylesForState:(TextInputState)state {
...
         return;
     }
+
+	UITextRange *range = self.selectedTextRange;
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextField+Styles.m]: align

File: Styles/Classes/UIElement+Extensions/UITextField+Styles.m [hunk 3]
```diff
@@ -132,10 +132,16 @@ - (void)updateStylesForState:(TextInputState)state {
...
     [self setAttributedText:[style applyTo:text]];
+	if (range) {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextField+Styles.m]: align

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 4]
```diff
@@ -127,7 +127,12 @@ - (void)applyTextStyle:(TextStyle *)style {
     if (!text || !style) {
         return;
     }
-    [self setAttributedText:[style applyTo:text]];
+
+	UITextRange *range = self.selectedTextRange;
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: align

File: Example/Styles/Sources/Scenes/UIElements/UIElementsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+extension 	UIElementsViewController: StylePickerDelegate {
+	func didPickTextStyleDefinition(_ definition: TextStyle.Definition) {
+		textStyle = definition.textStyle
+	}
+
+	func didPickViewStyleDefinition(_ definition: ViewStyle.Definition) {
+		viewStyle = definition.viewStyle
+
```
Comment (jakubpetrik) [inline Example/Styles/Sources/Scenes/UIElements/UIElementsViewController.swift]: empty line

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m (hunk 4 above)
Comment (radimhalfar) [reply Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: it is -- and all above too -- github shows that as invalid 🤷‍♂️

**Pull request merged**